pkg/apis/options/legacy_options.go

```diff
@@ -403,3 +424,192 @@ func getXAuthRequestAccessTokenHeader() Header {
+	flagSet.String("oidc-email-claim", providers.OIDCEmailClaim, "which OIDC claim contains the user's email")
+	flagSet.String("login-url", "", "Authentication endpoint")
+	flagSet.String("redeem-url", "", "Token redemption endpoint")
+	flagSet.String("profile-url", "", "Profile access endpoint")
+	flagSet.String("resource", "", "The resource that is protected (Azure AD only)")
+	flagSet.String("validate-url", "", "Access token validation endpoint")
+	flagSet.String("scope", "", "OAuth scope specification")
+	flagSet.String("prompt", "", "OIDC prompt")
+	flagSet.String("approval-prompt", "force", "OAuth approval_prompt")
+
+	flagSet.String("acr-values", "", "acr values string:  optional")
+	flagSet.String("jwt-key", "", "private key in PEM format used to sign JWT, so that you can say something like -jwt-key=\"${OAUTH2_PROXY_JWT_KEY}\": required by login.gov")
+	flagSet.String("jwt-key-file", "", "path to the private key file in PEM format used to sign the JWT so that you can say something like -jwt-key-file=/etc/ssl/private/jwt_signing_key.pem: required by login.gov")
+	flagSet.String("pubjwk-url", "", "JWK pubkey access endpoint: required by login.gov")
+
+	flagSet.String("user-id-claim", "email", "which claim contains the user ID")
+	flagSet.StringSlice("allowed-group", []string{}, "restrict logins to members of this group (may be given multiple times)")
+
+	return flagSet
+}
+
+func (l *LegacyProvider) convert() (Providers, error) {
+	providers := Providers{}
+
+	provider := Provider{
+		ClientID:          l.ClientID,
+		ClientSecret:      l.ClientSecret,
+		ClientSecretFile:  l.ClientSecretFile,
+		ProviderType:      l.ProviderType,
+		ProviderCAFiles:   l.ProviderCAFiles,
+		LoginURL:          l.LoginURL,
+		RedeemURL:         l.RedeemURL,
+		ProfileURL:        l.ProfileURL,
+		ProtectedResource: l.ProtectedResource,
+		ValidateURL:       l.ValidateURL,
+		Scope:             l.Scope,
+		Prompt:            l.Prompt,
+		ApprovalPrompt:    l.ApprovalPrompt,
+		AllowedGroups:     l.AllowedGroups,
+		AcrValues:         l.AcrValues,
+	}
+
+	// This part is out of the switch section for all providers that support OIDC
+	provider.OIDCConfig = OIDCOptions{
+		OIDCIssuerURL:                      l.OIDCIssuerURL,
+		InsecureOIDCAllowUnverifiedEmail:   l.InsecureOIDCAllowUnverifiedEmail,
+		InsecureOIDCSkipIssuerVerification: l.InsecureOIDCSkipIssuerVerification,
+		SkipOIDCDiscovery:                  l.SkipOIDCDiscovery,
+		OIDCJwksURL:                        l.OIDCJwksURL,
+		UserIDClaim:                        l.UserIDClaim,
+		OIDCEmailClaim:                     l.OIDCEmailClaim,
+		OIDCGroupsClaim:                    l.OIDCGroupsClaim,
+	}
+
+	// This part is out of the switch section because azure has a default tenant
+	// that needs to be added from legacy options
+	provider.AzureConfig = AzureOptions{
+		AzureTenant: l.AzureTenant,
+	}
+
+	switch provider.ProviderType {
```

Comment:
Thanks for this breakdown!

This will be super helpful when we try to breakout the organization and documentation of per-provider configs!